src/main/kotlin/systems/ajax/motrechko/airguardian/service/DeliveryOrderService.kt

```diff
@@ -3,62 +3,74 @@ package systems.ajax.motrechko.airguardian.service
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 import systems.ajax.motrechko.airguardian.enums.DeliveryStatus
 import systems.ajax.motrechko.airguardian.enums.DroneStatus
 import systems.ajax.motrechko.airguardian.exception.DeliveryOrderNotFoundException
 import systems.ajax.motrechko.airguardian.model.DeliveryOrder
-import systems.ajax.motrechko.airguardian.repository.DeliveryOrderMongoRepository
+import systems.ajax.motrechko.airguardian.repository.DeliveryOrderMongoReactiveRepository
 import systems.ajax.motrechko.airguardian.repository.DeliveryOrderRepository
+import systems.ajax.motrechko.airguardian.repository.DroneRepository
 
 @Service
 class DeliveryOrderService(
     private val deliveryOrderRepository: DeliveryOrderRepository,
-    private val deliveryOrderCustomRepository: DeliveryOrderMongoRepository,
-    private val droneService: DroneService,
-    private val droneLogisticsService: DroneLogisticsService
+    private val deliveryOrderCustomRepository: DeliveryOrderMongoReactiveRepository,
+    private val droneLogisticsService: DroneLogisticsService,
+    private val droneRepository: DroneRepository
 ) {
-    fun createNewOrder(order: DeliveryOrder): DeliveryOrder {
-        val availableDrones = droneLogisticsService.findAvailableDrones(order.items)
-
-        return if (availableDrones.isNotEmpty()) {
-            droneLogisticsService.initializeOrderForDelivery(order, availableDrones)
-            deliveryOrderRepository.save(order)
-            logger.info("find {} drones for delivery new order with id {}", availableDrones.size, order.id)
-            order
-        } else {
-            order.status = DeliveryStatus.IN_PROGRESS
-            deliveryOrderRepository.save(order)
-            logger.info("no free drones were found for prayer with id {}", order.id)
-            order
-        }
+    fun createNewOrder(order: DeliveryOrder): Mono<DeliveryOrder> {
+        return droneLogisticsService.findAvailableDrones(order.items)
+            .collectList()
+            .filter { availableDrones -> availableDrones.isNotEmpty() }
```

Comment:
```suggestion
            .filter { it.isNotEmpty() }
```
don't afraid to use 'it' when naming is not definitely needed. here it's easy to understand that you're checking that list is not empty